lib/grape/validations/params_scope.rb

```diff
@@ -293,6 +293,8 @@ def validates(attrs, validations)
         # type casted values
         coerce_type validations, attrs, doc_attrs, opts
 
+        move_alias_to_end(validations)
```

Comment:
This isn't reused, and we do a ton of things above, so I would just inline the two lines of the method here.

I think `delete` also always makes a copy, and we should assume a `nil` value is a thing, so to be consistent I what we do above:

```ruby
# explain why
validations[:as] = validations.delete(:as) if validations.key?(:as)
```

Reply:
Errrr.... re-reading the code we move that `:as` to the "end". I understand why it works, but feels to me that it's wrong to rely on the order in `validations`.

Reply:
I agree that relying on the order is not good, but that was the easier way I found to fix the bug so far.

Another approach would be set a variable to let others validators know about alias, but it seems to be a bigger change and I am not sure whether the outcome would be worth..

Please let me know whether you have more ideas.

Reply:
I just don't think we _can_ use on this approach to work reliably. There're no ordering guarantees. I think extracting `as` somehow is the proper way of doing it. Give it a try! We're here to help.

Reply:
Check the new approach now please. Build is failing because of rubocop `lib/grape/endpoint.rb:6:3: C: Class has too many lines. [295/288]`.

Reply:
Run `rubocop -a ; rubocop --auto-gen-config` for this.